A3A/addons/core/functions/Revive/fn_handleDamage.sqf

```diff
@@ -124,11 +123,27 @@ else
 			{
 				if !(_unit getVariable ["incapacitated",false]) then
 				{
-					[_unit, _injurer] call _makeUnconscious;
+					private _layer1 = cutText ["You've been critically hit, and are losing your senses soon...", "PLAIN", 2];
+					0 cutFadeOut 2;
+					[_unit, _injurer] spawn FN_DELAYUNCONSCIOUS;
 				};
 			};
 		};
 	};
 };
 
-_damage
+// GAMERARMY: Delays unconsciousnes by 5 seconds, giving the player a brief moment to escape the gun fire.
+// Good for 2-3 people team playthrough.
+FN_DELAYUNCONSCIOUS = {
+	params ["_unit", "_injurer"];
+
+	_unit allowDamage false;
+	_unit setDamage 0.7;
+	_unit allowDamage false;
+	private _duration = time + 5;
```

Comment:
Any reason for using this over ```sleep```? ```sleep``` should yield so the next script can be scheduled, whereas ```waitUntil```, does not yield and instead just [busy waits](https://en.wikipedia.org/wiki/Busy_waiting) until the condition is met.

If there is no specific reason for using ```waitUntil```, this would be better off as a ```sleep``` so that something else is allowed to run, instead of this script being given execution just to sit there spinning.